src/harness/certs/generate_fake_certs.sh

```diff
@@ -256,13 +235,86 @@ openssl ca -cert cbsd_ca.cert -keyfile private/cbsd_ca.key -in client_expired.cs
+    -out unrecognized_domain_proxy.cert -outdir ./root \
+    -policy policy_anything -extensions oper_req_sign -config ../../../cert/openssl.cnf \
+    -batch -notext -create_serial -utf8 -days 1185 -md sha384
+
+# Certificates for test case WINN.FT.S.SDS.7 - corrupted certificate, based on domain_proxy.cert
+echo "\n\nGenerate 'corrupted_domain_proxy' certificate/key"
+gen_corrupt_cert domain_proxy.key domain_proxy.cert corrupted_domain_proxy.key corrupted_domain_proxy.cert
+
+#Certificate for test case WINNF.FT.S.SDS.8 - Self-signed certificate presented during registration
+#Using the same CSR that was created for normal operation
+echo "\n\nGenerate 'self_signed_domain_proxy' certificate/key"
+openssl x509 -signkey domain_proxy.key -in domain_proxy.csr \
+    -out self_signed_domain_proxy.cert \
+    -req -days 1185
+
+#Certificate for test case WINNF.FT.S.SDS.9 - Non-CBRS trust root signed certificate presented during registration
+echo "\n\nGenerate non_cbrs_root_signed_oper_ca.csr certificate/key"
+
+openssl req -new -newkey rsa:4096 -nodes \
+    -reqexts oper_ca -config ../../../cert/openssl.cnf \
+    -out non_cbrs_root_signed_oper_ca.csr -keyout private/non_cbrs_root_signed_oper_ca.key \
+    -subj "/C=US/ST=District of Columbia/L=Washington/O=Wireless Innovation Forum/OU=www.wirelessinnovation.org/CN=WInnForum RSA Domain Proxy CA-2"
+
+openssl ca -cert non_cbrs_root_ca.cert -keyfile private/non_cbrs_root_ca.key -in non_cbrs_root_signed_oper_ca.csr \
+    -policy policy_anything -extensions oper_ca_sign -config ../../../cert/openssl.cnf \
+    -out non_cbrs_root_signed_oper_ca.cert -outdir ./root \
+    -batch -notext -create_serial -utf8 -days 5475 -md sha384
+
+#Generate a Domain Proxy certifcate signed by an intermediate Domain Proxy CA which is signed by a non-CBRS root CA
+echo "\n\nGenerate domain_proxy certificate/key"
+openssl req -new -newkey rsa:2048 -nodes \
+    -reqexts oper_req -config ../../../cert/openssl.cnf \
+    -out non_cbrs_signed_domain_proxy.csr -keyout non_cbrs_signed_domain_proxy.key \
+    -subj "/C=US/ST=District of Columbia/L=Washington/O=Wireless Innovation Forum/OU=www.wirelessinnovation.org/CN=Domain Proxy Unknown"
+openssl ca -cert non_cbrs_root_signed_oper_ca.cert -keyfile private/non_cbrs_root_signed_oper_ca.key -in non_cbrs_signed_domain_proxy.csr \
+    -out non_cbrs_signed_domain_proxy.cert -outdir ./root \
+    -policy policy_anything -extensions oper_req_sign -config ../../../cert/openssl.cnf \
+    -batch -notext -create_serial -utf8 -days 1185 -md sha384
+
+# Certificate for test case WINNF.FT.S.SDS.10 - Certificate of wrong type presented during registration
+# Creating a wrong type certificate by reusing the server.csr and having it signed by proxy_ca.
+echo "\n\nGenerate wrong type certificate/key"
+openssl ca -cert proxy_ca.cert -keyfile private/proxy_ca.key -in server.csr \
```

Comment:
The way the test case is written the certificate is signed by cbsd_sa, so wouldn’t the SAS then accept the certificate and not fail to establish the TLS session?  Should the certificate be signed by the sas_ca instead in this TC?

Reply:
Agreed with you according to teh following statement. Will make the change.

"Replace the DP Test Harness certificate with an invalid certificate. This certificate shall be valid except for being a certificate of the wrong type but having correct signatures descending from the CBRS trust root (e.g. a SAS certificate, not a CBSD certificate)"